tests/mpls/test_mpls.py

```diff
@@ -0,0 +1,594 @@
+import os
+import time
+import random
+import logging
+import pprint
+import json
+
+from abc import ABCMeta, abstractmethod
+
+import pytest
+
+import ptf.testutils as testutils
+import ptf.mask as mask
+import ptf.packet as packet
+
+from tests.common import reboot, port_toggle
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer, LogAnalyzerError
+from tests.common.fixtures.duthost_utils import backup_and_restore_config_db_module
+from tests.common.config_reload import config_reload
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+#    pytest.mark.disable_loganalyzer,  # disable automatic loganalyzer
+    pytest.mark.topology('t1'),
+#    pytest.mark.sanity_check(skip_sanity=True)
+]
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+DUT_TMP_DIR='/tmp'
+ADD_DIR = os.path.join(BASE_DIR, 'configs')
+
+LABEL_POP_ROUTES='label_pop_routes'
+LABEL_PUSH_ROUTES='label_push_routes'
+LABEL_SWAP_ROUTES='label_swap_routes'
+LABEL_DEL_ROUTES='label_del_routes'
+
+@pytest.fixture(scope='module')
+def setup(duthost, tbinfo, ptfadapter):
+    """
+    setup fixture gathers all test required information from DUT facts and tbinfo
+    :param duthost: DUT host object
+    :param tbinfo: fixture provides information about testbed
+    :return: dictionary with all test required information
+    """
+    # gather ansible facts
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    int_facts = duthost.interface_facts()['ansible_facts']
+    host_facts = duthost.setup()['ansible_facts']
+
+    tor_ports_ids=[]
+    tor_ports={}
+    spine_ports_ids=[]
+    spine_ports={}
+    tor_addr={}
+    tor_peer_addr={}
+    spine_addr={}
+    spine_peer_addr={}
+    tor_mac={}
+    spine_mac={}
+ 
+    # get the list of TOR/SPINE ports
+    for dut_port, neigh in mg_facts['minigraph_neighbors'].items():
+        if not port_up(duthost, dut_port): 
+            continue
+        port_id = mg_facts['minigraph_port_indices'][dut_port]
+        if 'T0' in neigh['name']:
+            tor_ports[port_id]=dut_port
+            tor_ports_ids.append(port_id)
+        elif 'T2' in neigh['name']:
+            spine_ports[port_id]=dut_port
+            spine_ports_ids.append(port_id)
+    
+    logger.info('tor_ports: {}'.format(tor_ports))
+    logger.info('spine_ports: {}'.format(spine_ports))    
+        
+    for blk in mg_facts['minigraph_interfaces']:
```

Comment:
some t1 topologies have only Port Channel interfaces, this will not work. But if you use the 'get_active_ip_interfaces' there should be enough information to build the test parameters.